### DISCUSSION TITLE
How to use the apfs-types crate

### DISCUSSION THREAD

**Discussion by theoparis** [indygreg/apple-platform-rs#165] [Ideas]
Does anyone have an example of using apfs-types to manage an apfs filesystem, either on a physical disk or a file on an existing file system? I would create an example myself but I'm yet to figure out how the on disk structure (e.g. superblock) in general works... Maybe it's time for me to learn that 😅

**Comment by indygreg** [comment]
Check out the `apfs-next` branch (https://github.com/indygreg/apple-platform-rs/tree/apfs-next). The `apfs-core` and `apfs-cli` crates implement a very basic reader/writer interface.

I haven't merged the code because things aren't very well tested. But the code can hopefully point you in the right direction.

Many of APFS's internals are under-documented. Good luck.
